### ISSUE TITLE
SimpleStatisticsCollector throws ConcurrentModificationException

### ISSUE THREAD

**Issue by jeggy** [nidomiro/KDataLoader#13]
Sometimes with a huge data tree I'm getting this error:
```
java.util.ConcurrentModificationException: null
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:1013) ~[?:?]
	at java.util.ArrayList$Itr.next(ArrayList.java:967) ~[?:?]
	at nidomiro.kdataloader.statistics.SimpleStatisticsCollector.createStatisticsSnapshot(SimpleStatisticsCollector.kt:80) ~[KDataLoader-jvm-0.3.0.jar:?]
	at nidomiro.kdataloader.statistics.SimpleStatisticsCollector$createStatisticsSnapshot$1.invokeSuspend(SimpleStatisticsCollector.kt) ~[KDataLoader-jvm-0.3.0.jar:?]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) [kotlin-stdlib-1.5.0.jar:1.5.0-release-749 (1.5.0)]
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) [kotlinx-coroutines-core-jvm-1.4.3-native-mt.jar:?]
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571) [kotlinx-coroutines-core-jvm-1.4.3-native-mt.jar:?]
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750) [kotlinx-coroutines-core-jvm-1.4.3-native-mt.jar:?]
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678) [kotlinx-coroutines-core-jvm-1.4.3-native-mt.jar:?]
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665) [kotlinx-coroutines-core-jvm-1.4.3-native-mt.jar:?]
```

This happens inside the KGraphQL when having multiple thousands of records and where it's using nested GraphQL fields.

**Comment by jeggy** [comment]
I'm reporting it here now. But I will try and help or give more information about how to reproduce soon

**Comment by nidomiro** [comment]
I made the `SimpleStatisticsCollector` more simplistic, may be it's fixed in 0.5.0 - if not, feel free to reopen
